cloudinit/util.py

```diff
@@ -3094,6 +3094,14 @@ def wait_for_files(flist, maxwait, naplen=0.5, log_pre=""):
     return need
 
 
+def wait_for_snap_seeded():
```

Comment:
This approach results in the same function call, which uses the same magic string and callback in multiple configuration modules.


Something like this would be cleaner I think:
```python
def wait_for_snap_seeded():
    """Helper to wait on completion of snap seeding."""
    def callback():
        if not subp.which("snap"):
            LOG.debug("Skipping snap wait, no snap command present")
            return
        subp.subp(["snap", "wait", "system", "seed.loaded"])
    cloud.run("snap-seeded", callback, [], freq=PER_ONCE)
```

that would turn our various calls like this:
```python
    cloud.run("snap-seeded", util.wait_for_snap_seeded, [], freq=PER_ONCE)
```

into this:
```python
    util.wait_for_snap_seeded()
```

Reply:
Thanks @holmanb for the review. What this suggestion doesn't capture though is we would also need to provide the cloud as a param into `wait_for_snap_seeded`. I had wanted to keep dependency util function on Cloud instances to a minimal in `util` just for separation of roles/functions of util module functions and leaving this function as one that could be called outside the scope of an environment that has access to a `Cloud` instance. If we tightly couple the `wait_for_snap_seeded`  function to the presence of the cloud instance, it strictly limits call-sites to having access to the `Cloud` instance.   Maybe that dependency is ok though as we have no other call-sites in mind today.

Reply:
@holmanb my defense aside. I can't immediately think of other cases where we may intend as call-sites of `wait_for_snap_seeded` outside a scope where `cloud` is already present. So, let's design for today's call-sites. If we have to revist for other use-cases we'll be better placed/aware of a design that incorporates both needs. And it's trivial to fix that if we ever need to.